File: annotation-processor-common/src/testFixtures/java/ru/tinkoff/kora/annotation/processor/common/TestAspect.java
package ru.tinkoff.kora.annotation.processor.common;

import ru.tinkoff.kora.common.AopAnnotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@AopAnnotation
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface TestAspect {

}

File: symbol-processor-common/src/testFixtures/kotlin/ru/tinkoff/kora/ksp/common/TestAspect.kt
package ru.tinkoff.kora.ksp.common

import com.google.devtools.ksp.processing.Resolver
import com.google.devtools.ksp.symbol.KSFunctionDeclaration
import ru.tinkoff.kora.aop.symbol.processor.KoraAspect
import ru.tinkoff.kora.aop.symbol.processor.KoraAspectFactory
import ru.tinkoff.kora.common.AopAnnotation
import java.util.*

@AopAnnotation
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class TestAspect

class TestAspectKoraAspect : KoraAspect {

    override fun getSupportedAnnotationTypes(): Set<String> = setOf(TestAspect::class.qualifiedName!!)

    override fun apply(ksFunction: KSFunctionDeclaration, superCall: String, aspectContext: KoraAspect.AspectContext): KoraAspect.ApplyResult = KoraAspect.ApplyResult.Noop.INSTANCE
}

class TestAspectKoraAspectFactory : KoraAspectFactory {
    override fun create(resolver: Resolver) = TestAspectKoraAspect()
}
